src/components/shared/TreeView.js

```diff
@@ -58,22 +58,35 @@ const TreeViewHeader = <DisplayData: Object>({
   }
   return (
     <div className="treeViewHeader">
-      {fixedColumns.map((col) => (
-        <Localized
-          id={col.titleL10nId}
-          attrs={{ title: true }}
-          key={col.propName}
-        >
+      {fixedColumns.map((col) =>
+        col.titleL10nId ? (
+          <Localized
+            id={col.titleL10nId}
+            attrs={{ title: true }}
+            key={col.propName}
+          >
+            <span
+              className={`treeViewHeaderColumn treeViewFixedColumn ${col.propName}`}
+            ></span>
+          </Localized>
+        ) : (
           <span
+            key={col.propName}
             className={`treeViewHeaderColumn treeViewFixedColumn ${col.propName}`}
           ></span>
+        )
+      )}
```

Comment:
I wasn't sure about a new component because there could be some edge cases and I wanted to add extensive tests for them. Like sometimes we would need to propagate the props to children and that will not be possible with this component. I'm still a bit skeptical about it, but I can add it to this file then since this is directly what we do in the code. Could you take a look at it again?

Reply:
I'm not sure if this is a valid case, but I was thinking about propagating the prop so we pass to children directly. Something like this:
```js
function PermissiveLocalized(props: React.ElementConfig<typeof Localized>) {
  const { children, id, ...otherProps } = props;
  return id ? (
    <Localized {...props}>{children}</Localized>
  ) : (
    React.cloneElement(children, { ...otherProps })
  );
}
```

See the else branch in the ternary. So we can do something like: "put this prop to the first element, either that first one is the Localized or children. But I'm not sure if that makes sense at all. And probably we can workaround this easily.

Reply:
I could see that could be useful if we want to pass a fallback content to the child in case it's not localized. But in that case I'd prefer having 2 sets of prop, maybe: `{ localizedProps, contentIfNotLocalized }`.

But we can as well pass it always to the child, and if it's localized that will be replaced...